static/app/views/replays/list/replaysList.tsx

```diff
@@ -45,55 +45,106 @@ function ReplaysList() {
     );
   }, [location]);
 
-  const {replays, pageLinks, isFetching, fetchError} = useReplayList({
-    eventView,
-    location,
-    organization,
-  });
+  return <ReplaysListFetcher eventView={eventView} location={location} />;
+}
+
+function ReplaysListFetcher({
+  eventView,
+  location,
+}: {
+  eventView: EventView;
+  location: Location;
+}) {
+  const organization = useOrganization();
 
+  const hasSessionReplay = organization.features.includes('session-replay');
   const {hasSentOneReplay, fetching} = useHaveSelectedProjectsSentAnyReplayEvents();
 
+  if (hasSessionReplay && !fetching && hasSentOneReplay) {
+    return (
+      <ReplaysListTable
+        eventView={eventView}
+        location={location}
+        organization={organization}
+      />
+    );
+  }
+  return (
+    <Page>
+      <ReplayOnboardingPanel />
+    </Page>
+  );
+}
+
+function Page({children}: {children: ReactNode}) {
+  const organization = useOrganization();
+
   return (
     <Layout.Body>
       <Layout.Main fullWidth>
         <ReplaysFilters />
-        {fetching ? null : hasSentOneReplay ? (
-          <Fragment>
-            <ReplayTable
-              fetchError={fetchError}
-              isFetching={isFetching}
-              replays={replays}
-              sort={eventView.sorts[0]}
-              visibleColumns={[
-                ReplayColumns.session,
-                ...(hasRoomForColumns
-                  ? [ReplayColumns.projectId, ReplayColumns.startedAt]
-                  : []),
-                ReplayColumns.duration,
-                ReplayColumns.countErrors,
-                ReplayColumns.activity,
-              ]}
-            />
-            <Pagination
-              pageLinks={pageLinks}
-              onCursor={(cursor, path, searchQuery) => {
-                trackAdvancedAnalyticsEvent('replay.list-paginated', {
-                  organization,
-                  direction: cursor?.endsWith(':1') ? 'prev' : 'next',
-                });
-                browserHistory.push({
-                  pathname: path,
-                  query: {...searchQuery, cursor},
-                });
-              }}
-            />
-          </Fragment>
-        ) : (
-          <ReplayOnboardingPanel />
-        )}
+        <Feature
+          features={['session-replay']}
+          organization={organization}
+          renderDisabled={() => <ReplayOnboardingPanel />}
+        >
+          {children}
+        </Feature>
       </Layout.Main>
     </Layout.Body>
   );
 }
 
+function ReplaysListTable({
+  eventView,
+  location,
+  organization,
+}: {
+  eventView: EventView;
+  location: Location;
+  organization: Organization;
+}) {
+  const theme = useTheme();
+  const hasRoomForColumns = useMedia(`(min-width: ${theme.breakpoints.small})`);
+
+  const {replays, pageLinks, isFetching, fetchError} = useReplayList({
+    eventView,
+    location,
+    organization,
+  });
+
+  return (
+    <Page>
```

Comment:
nit: you have `ReplaysListFetcher` wrapping `ReplayOnboardingPanel` with `Page`, but have `Page` here inside of `ReplayListTable`.

Reply:
i think now that `ReplaysListFetcher` isn't doing much. 

I tore it apart because the `useReplayList` was happening unconditionally, it's in too many bits right now i think

Reply:
woah! so much better now. `<Page>` is no longer a thing ,but there's no duplicate of that stuff.

And just one inner component inside this file.